src/com/rcv/CastVoteRecord.java

```diff
@@ -1,23 +1,50 @@
 package com.rcv;
 
-import com.fasterxml.jackson.annotation.JsonAnySetter;
 
 import java.util.*;
 
-// map of contest IDs to ContestRankings objects
-// i.e. a Ballot
+// internal representation of a cast vote record
+// contains an ID
+// contains a mapping of rank (integer) to a set of candidate ID(s) selected at that rank
+// a set is used to handle overvotes
+
 public class CastVoteRecord {
-  private Map<Integer, ContestRankings> rankings = new HashMap<Integer, ContestRankings>();
+  public String mBallotID;
+  List<ContestRanking> mRankings;
+  SortedMap<Integer, Set<String>> mSortedRankings;
 
-  @JsonAnySetter
-  public void add(String key, ContestRankings value) {
-    Integer electionId = Integer.parseInt(key);
-    rankings.put(electionId, value);
-  }
+  // mDescriptionsByRound contains who this ballot counted for in each round
+  // followed by reason for exhaustion if it is ever exhausted
+  public List<String> mDescriptionsByRound = new ArrayList<>();
```

Comment:
I recognize that technically an ArrayList can work here, but I'd rather use a map from Integer to String to make it explicit which round maps to which description. And then we wouldn't have to reference the round directly in the description string, and could instead store it in a structured way.

Reply:
done